### PR TITLE
refactor: PPS-82 PPS-84 major refactoring for crawl and exporter tests

### PR DESCRIPTION
**Pull request by Luca1313**
refactor to remove majority of warnings, refactored methods name, provided export with DSL test and a skeleton for full application test with DSL and standard config

### CODE REVIEW

File: src/main/scala/core/scraper/Aggregator.scala [hunk 1]
```diff
@@ -75,6 +75,8 @@ object Aggregator:
               val1
             case (_, Some(val2)) =>
               val2
+            case _ =>
+              throw new IllegalStateException("An unexpected error occurs.")
```
Comment (giovaz94) [inline src/main/scala/core/scraper/Aggregator.scala]: Is it really necessary using Exceptions here?

File: src/test/scala/utility/MockServer.scala
Comment (v4l3rio) [inline src/test/scala/utility/MockServer.scala:1]: the prints in this file should be removed

**Review by giovaz94 — COMMENTED** [review]
Really nice work!
The only consideration I have found is the use of an exception in the code, but if it's _really_ necessary  i guess it's ok for me

**Review by FreshMag — APPROVED** [review]

File: src/main/scala/core/scraper/Aggregator.scala (hunk 1 above)
Comment (FreshMag) [reply src/main/scala/core/scraper/Aggregator.scala]: I agree, there should probably be a different design

**Review by giovaz94 — APPROVED** [review]

**Review by FreshMag — APPROVED** [review]

File: src/test/scala/utility/MockServer.scala
Comment (Luca1313) [reply src/test/scala/utility/MockServer.scala:1]: Done

File: src/main/scala/core/scraper/Aggregator.scala (hunk 1 above)
Comment (Luca1313) [reply src/main/scala/core/scraper/Aggregator.scala]: By means of unchecked tag I avoided the usage of the exception

**Pull request merged**